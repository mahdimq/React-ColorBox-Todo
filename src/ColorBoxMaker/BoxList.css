.BoxList {
	display: flex;
	justify-content: center;
	align-items: center;
	margin: 2em;
	flex-wrap: wrap;
}
