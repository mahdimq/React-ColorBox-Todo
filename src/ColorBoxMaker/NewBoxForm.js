import React, { useState } from 'react';

function NewBoxForm({ addBox }) {
	const INITIAL_STATE = {
		width: '',
		height: '',
		backgroundColor: ''
	};

	const [formData, setFormData] = useState(INITIAL_STATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addBox({ ...formData });
		setFormData(INITIAL_STATE);
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor='width'>Width</label>
			<input
				id='width'
				type='number'
				name='width'
				min={0}
				placeholder='Box Width'
				value={formData.width}
				onChange={handleChange}
			/>
			<label htmlFor='height'>Height</label>
			<input
				id='height'
				type='number'
				name='height'
				min={0}
				placeholder='Box Height'
				onChange={handleChange}
				value={formData.height}
			/>
			<label htmlFor='backgroundColor'>Background Color</label>
			<input
				id='backgroundColor'
				type='text'
				name='backgroundColor'
				placeholder='Background Color'
				onChange={handleChange}
				value={formData.backgroundColor}
			/>
			<button>Add Box</button>
		</form>
	);
}

export default NewBoxForm;
