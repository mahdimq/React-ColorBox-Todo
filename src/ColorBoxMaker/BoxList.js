import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import './BoxList.css';

function BoxList() {
	const INTIAL_STATE = [
		{
			id: uuid(),
			width: 100,
			height: 100,
			backgroundColor: 'red'
		},
		{
			id: uuid(),
			width: 100,
			height: 100,
			backgroundColor: 'orange'
		},
		{
			id: uuid(),
			width: 100,
			height: 100,
			backgroundColor: 'slateblue'
		}
	];
	const [boxSpecs, setBoxSpecs] = useState(INTIAL_STATE);
	const addBox = (newBox) => {
		setBoxSpecs((boxSpecs) => [...boxSpecs, { id: uuid(), ...newBox }]);
	};

	const handleDelete = (box) => {
		setBoxSpecs(boxSpecs.filter((val) => val !== box));
	};

	return (
		<>
			<div className='BoxList'>
				{boxSpecs.map((specs) => (
					<Box
						backgroundColor={specs.backgroundColor}
						width={specs.width}
						height={specs.height}
						key={specs.id}
						deleteBox={() => handleDelete(specs)}
					/>
				))}
			</div>
			<NewBoxForm addBox={addBox} />
		</>
	);
}

export default BoxList;
