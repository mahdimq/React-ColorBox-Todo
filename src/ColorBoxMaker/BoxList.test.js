import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';
import NewBoxForm from './NewBoxForm';

// SMOKE TEST
it('renders component without crashing', () => {
	render(<BoxList />);
});

// SNAPSHOT TEST
it('matches snapshot', () => {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});

// it('should add new box', () => {
// 	const { queryByText, getByLabelText } = render(<BoxList />);
// 	const input = getByLabelText('Width');
// 	const input = getByLabelText('Background Color');
// 	const button = queryByText('Add Box');

// 	expect(queryByText('Width: 100px')).not.toBeInTheDocument();
// 	fireEvent.change(input, { target: { value: '100px' } });
// 	fireEvent.click(button);
// 	expect(queryByText('Width: 100px')).toBeInTheDocument();
// });
