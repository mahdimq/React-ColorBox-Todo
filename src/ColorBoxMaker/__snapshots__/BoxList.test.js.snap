// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div
      class="Box"
      style="background-color: red; width: 100px; height: 100px;"
    >
      <button>
        X
      </button>
    </div>
    <div
      class="Box"
      style="background-color: orange; width: 100px; height: 100px;"
    >
      <button>
        X
      </button>
    </div>
    <div
      class="Box"
      style="background-color: slateblue; width: 100px; height: 100px;"
    >
      <button>
        X
      </button>
    </div>
  </div>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      min="0"
      name="width"
      placeholder="Box Width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      min="0"
      name="height"
      placeholder="Box Height"
      type="number"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Background Color"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
